packages/flutter/test/widgets/app_lifecycle_listener_test.dart

```diff
@@ -129,25 +129,20 @@ void main() {
     await setAppLifeCycleState(AppLifecycleState.resumed);
     expect(transitions, equals(<String>['restart', 'show', 'resume']));
 
-    // Generates intermediate states.
+    // Generates intermediate states from lower to higher lifecycle states
     transitions.clear();
     await setAppLifeCycleState(AppLifecycleState.paused);
     expect(transitions, equals(<String>['inactive', 'hide', 'pause']));
-    // Wraps around from pause to detach.
-    await setAppLifeCycleState(AppLifecycleState.detached);
-    expect(transitions, equals(<String>['inactive', 'hide', 'pause', 'detach']));
```

Comment:
Why are these other cases removed?

We need to be able to generate all of the states in between the current one and the new state, even if it wraps around.

So, even though the web has no `hidden` state currently, we want to enable people to write code that does things when the app is hidden because it's more portable that way. On the web, this code would only get executed on the way to detached, but it would get executed. If we skip that state on the web, then the code would never execute there.

Reply:
I removed the test because, at the very end, I triggered a `detached` state. However, you are right the test should cover wrap-around scenarios.

The web platform does have a [`hidden`](https://github.com/flutter/engine/blob/c4247c5e31babcf8512e156fec1af447fae1c057/lib/web_ui/lib/src/engine/platform_dispatcher/app_lifecycle_state.dart#L103) state, but it is only activated when the tab is changed. This means it's possible for the app to enter a `detached` state (be killed) when focus is lost.